src/services/permissions-checker.js

```diff
@@ -16,58 +16,144 @@ class PermissionsChecker {
+    if (PermissionsChecker.permissions.renderings[this.renderingId]
+      && PermissionsChecker.permissions.renderings[this.renderingId].data
+      && PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName]) {
+      return PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName].scope;
+    }
+    return null;
+  }
+
+  static transformPermissionsFromOldToNewFormat(permissions) {
+    Object.keys(permissions).forEach((modelName) => {
+      const { collection } = permissions[modelName];
+      permissions[modelName].collection = {
+        browseEnabled: collection.list || collection.searchToEdit,
+        readEnabled: collection.show,
+        editEnabled: collection.create,
+        addEnabled: collection.update,
+        deleteEnabled: collection.delete,
+        exportEnabled: collection.export,
+      };
+
+      if (permissions[modelName].actions) {
+        Object.keys(permissions[modelName].actions).forEach((actionName) => {
+          const action = permissions[modelName].actions[actionName];
+          permissions[modelName].actions[actionName] = {
+            triggerEnabled: action.users ? action.allowed && action.users : action.allowed,
+          };
+        });
+      }
+    });
+
+    return permissions;
   }
 
-  static getLastRetrieveTime(renderingId) {
-    const permissions = PermissionsChecker.getPermissions(renderingId);
-    return permissions ? permissions.lastRetrieve : null;
+  _setTeamsACLPermissions(permissions) {
+    const newFormatPermissions = permissions
+      ? PermissionsChecker.transformPermissionsFromOldToNewFormat(permissions)
+      : null;
+
+    PermissionsChecker.permissions.renderings[this.renderingId] = {
+      data: newFormatPermissions,
+      lastRetrieve: moment(),
+    };
+  }
+
+  _setRolesACLPermissions(permissions) {
+    PermissionsChecker.permissions.collections = {
+      data: permissions.collections,
+      lastRetrieve: moment(),
+    };
+    PermissionsChecker.permissions.renderings[this.renderingId] = {
+      data: permissions.renderings ? permissions.renderings[this.renderingId] : null,
+      lastRetrieve: moment(),
+    };
+  }
+
+  _setPermissions(permissions) {
+    if (PermissionsChecker.isrolesACLActivated) {
+      this._setRolesACLPermissions(permissions);
+    } else {
+      this._setTeamsACLPermissions(permissions);
+    }
+  }
+
+  static getLastRetrieveTime(renderingId, permissionName) {
+    // In the case of rolesACL format and browseEnabled permission, the last retrieve to be taken
+    // into account is the one stored by rendering (because of the scope information).
+    if (PermissionsChecker.isrolesACLActivated && permissionName !== 'browseEnabled') {
+      return PermissionsChecker.permissions.collections.lastRetrieve;
+    }
+    return PermissionsChecker.permissions.renderings[renderingId]
+      ? PermissionsChecker.permissions.renderings[renderingId].lastRetrieve
+      : null;
```

Comment:
This is due to the fact that you use 2 different formats when storing your data. You should not change this function